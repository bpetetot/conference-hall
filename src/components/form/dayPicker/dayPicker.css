.SingleDatePickerInput {
  background: none;
  border: 1px solid rgba(0, 0, 0, 0.3);
  display: flex;
  align-items: center;
  background: #fff;
  border-radius: 3px;
}

.SingleDatePickerInput .SingleDatePickerInput_calendarIcon {
  color: var(--primary-color);
  margin: 0;
}

.SingleDatePickerInput .DateInput {
  background: none;
  padding: 0;
  margin: 0;
  border: none;
  width: 100%;
  font-weight: normal;
  font-size: 1em;
  line-height: initial;
  color: var(--primary-color);
}

.SingleDatePickerInput .DateInput .DateInput_input {
  font-size: 1em;
  background: #fff;
  font-weight: inherit;
  border: none;
  color: #000;
  font-family: inherit;
  outline: none;
  padding: 11px 0 9px 0;
  transition: all 0.5s;
  width: 100%;
  box-shadow: none;
  cursor: pointer;
}

.SingleDatePickerInput .DateInput .DateInput_input:focus {
  background: #fff;
}

.SingleDatePicker_picker {
  z-index: 2;
}

.SingleDatePicker_picker__openDown {
  top: 50px;
}
.SingleDatePicker_picker .CalendarDay__selected {
  background: var(--secondary-color);
  border: 1px solid var(--secondary-color);
  color: #fff;
}

.SingleDatePicker_picker .DayPickerNavigation_button {
  border: none;
  background: none;
}
