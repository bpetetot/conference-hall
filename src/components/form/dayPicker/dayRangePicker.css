.DateRangePickerInput {
  background: none;
  border: 1px solid rgba(0, 0, 0, 0.3);
  display: flex;
  align-items: center;
  background: #fff;
  border-radius: 3px;
}

.DateRangePickerInput .DateRangePickerInput_calendarIcon {
  color: var(--primary-color);
  margin: 0;
}

.DateRangePickerInput .DateInput {
  background: none;
  padding: 0;
  margin: 0;
  border: none;
  width: 100%;
  font-weight: normal;
  font-size: 1em;
  line-height: initial;
  color: var(--primary-color);
}

.DateRangePickerInput .DateInput .DateInput_input {
  font-size: 1em;
  background: #fff;
  font-weight: inherit;
  border: none;
  color: #000;
  font-family: inherit;
  outline: none;
  padding: 11px 0 9px 0;
  transition: all 0.5s;
  width: 100%;
  box-shadow: none;
  cursor: pointer;
}

.DateRangePickerInput .DateInput .DateInput_input:focus {
  background: #fff;
}

.DateRangePickerInput .DateRangePickerInput_arrow {
  color: #000;
  margin: 0 1em;
}

.DateRangePicker_picker {
  z-index: 2;
}

.DateRangePicker_picker__openDown {
  top: 50px;
}

.DateRangePicker_picker .CalendarDay__selected {
  background: var(--secondary-color);
  border: 1px solid var(--secondary-color);
  color: var(--secondary-color-font);
}

.DateRangePicker_picker .CalendarDay__selected_span {
  background: var(--secondary-color-light);
  border: 1px solid var(--secondary-color-light);
  color: var(--secondary-color-font);
}

.DateRangePicker_picker .CalendarDay__hovered_span {
  background: var(--secondary-color-light);
  border: 1px solid var(--secondary-color-light);
  color: var(--secondary-color-font);
}

.DateRangePicker_picker .DayPickerNavigation_button {
  border: none;
  background: none;
}
