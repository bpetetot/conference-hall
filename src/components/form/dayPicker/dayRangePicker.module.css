.dayRangePicker {
  display: flex;
  align-items: center;
}

.arrow {
  margin: 0 1em;
}
