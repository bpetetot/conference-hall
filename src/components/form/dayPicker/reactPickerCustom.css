.day-picker-custom {
  border-radius: 3px;
  background: #fff;
  border: 1px solid var(--grey-color);
}

.day-picker-custom .react-datepicker__input-container {
  cursor: pointer;
}

.day-picker-custom  .react-datepicker__triangle::before {
  border-bottom-color: var(--grey-color) !important;
}

.day-picker-custom .react-datepicker__triangle {
  border-bottom-color: #fff !important;
}

.day-picker-custom .react-datepicker__header {
  background-color: #fff;
  border-bottom: none;
  padding-top: 1rem;
}

.day-picker-custom .react-datepicker__navigation {
  top: 1rem;
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.day-picker-custom .react-datepicker__day-names {
  margin: 0.5rem 1rem 0 1rem;
}

.day-picker-custom .react-datepicker__day-name {
  color: var(--grey-color-dark);
}

.day-picker-custom .react-datepicker__day--outside-month {
  color: var(--grey-color);
}

.day-picker-custom .react-datepicker__day-name,
.react-datepicker__day,
.react-datepicker__time-name {
  width: 2.5rem;
  line-height: 2.5rem;
  margin: 0;
}

.day-picker-custom .react-datepicker__month {
  margin: 0 1rem 1rem 1rem;
}

.day-picker-custom .react-datepicker__day:hover {
  border-radius: 0;
}

.day-picker-custom .react-datepicker__day--selected {
  border-radius: 0;
}

.day-picker-custom .react-datepicker__day--in-range {
  border-radius: 0;
  opacity: 0.8;
}

.day-picker-custom .react-datepicker__day--in-selecting-range {
  border-radius: 0;
}

@media (max-width: 400px), (max-height: 550px) {
  .day-picker-custom .react-datepicker__portal .react-datepicker__day-name,
  .react-datepicker__portal .react-datepicker__day,
  .react-datepicker__portal .react-datepicker__time-name {
    width: 2.5rem;
    line-height: 2.5rem;
  }
}
