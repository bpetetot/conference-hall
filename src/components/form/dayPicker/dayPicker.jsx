import React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'
import { SingleDatePicker } from 'react-dates'
import { toDate } from 'helpers/firebase'

import './dayPicker.css'

class DayPicker extends React.Component {
  constructor(props) {
    super(props)
    const initialValue = toDate(props.value)
    this.state = {
      focused: false,
      date: initialValue ? moment(initialValue) : undefined,
    }
  }

  onDateChange = (date) => {
    this.setState({ date })
    this.props.onChange(date ? date.toDate() : undefined)
  }

  onFocusChange = ({ focused }) => {
    this.setState({ focused })
  }

  render() {
    const { focused, date } = this.state
    const { id } = this.props
    return (
      <SingleDatePicker
        id={id}
        date={date}
        focused={focused}
        onDateChange={this.onDateChange}
        onFocusChange={this.onFocusChange}
        numberOfMonths={1}
        hideKeyboardShortcutsPanel
        readOnly
        displayFormat="MMMM Do YYYY"
        isOutsideRange={() => false}
        customInputIcon={<i className="fa fa-calendar" />}
      />
    )
  }
}

DayPicker.propTypes = {
  id: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.oneOfType([PropTypes.object, PropTypes.string, PropTypes.number]),
}

DayPicker.defaultProps = {
  value: undefined,
}

export default DayPicker
