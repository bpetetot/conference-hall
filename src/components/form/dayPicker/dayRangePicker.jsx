import React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'
import { DateRangePicker } from 'react-dates'
import { toDate } from 'helpers/firebase'

import './dayRangePicker.css'

let START_DATE = 'START_DATE_ID'
let END_DATE = 'END_DATE_ID'

class DayRangePicker extends React.Component {
  constructor(props) {
    super(props)
    const { value = {} } = props
    const initialStartDate = toDate(value.start)
    const initialEndDate = toDate(value.end)

    this.state = {
      focusedInput: undefined,
      start: initialStartDate ? moment(initialStartDate) : undefined,
      end: initialEndDate ? moment(initialEndDate) : undefined,
    }
  }

  componentWillMount() {
    const { id } = this.props
    START_DATE += `${id}Start`
    END_DATE += `${id}End`
  }

  onDatesChange = ({ startDate, endDate }) => {
    this.setState({ start: startDate, end: endDate })
    this.props.onChange({
      start: startDate ? startDate.toDate() : undefined,
      end: endDate ? endDate.toDate() : undefined,
    })
  }

  onFocusChange = (focusedInput) => {
    this.setState({ focusedInput })
  }

  render() {
    const { focusedInput, start, end } = this.state
    return (
      <div>
        <DateRangePicker
          startDateId={START_DATE}
          startDatePlaceholderText="Start Date"
          endDateId={END_DATE}
          endDatePlaceholderText="End Date"
          numberOfMonths={1}
          hideKeyboardShortcutsPanel
          readOnly
          displayFormat="MMMM Do YYYY"
          onDatesChange={this.onDatesChange}
          onFocusChange={this.onFocusChange}
          focusedInput={focusedInput}
          startDate={start}
          endDate={end}
          minimumNights={0}
          isOutsideRange={() => false}
          customInputIcon={<i className="fa fa-calendar" />}
          customArrowIcon={<i className="fa fa-arrow-right" />}
        />
      </div>
    )
  }
}

DayRangePicker.propTypes = {
  id: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.object, // eslint-disable-line react/forbid-prop-types
}

DayRangePicker.defaultProps = {
  value: undefined,
}

export default DayRangePicker
