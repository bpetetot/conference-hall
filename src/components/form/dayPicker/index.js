import 'react-datepicker/dist/react-datepicker-cssmodules.css'
import './reactPickerCustom.css'

export { default as DayPicker } from './dayPicker'
export { default as DayRangePicker } from './dayRangePicker'
