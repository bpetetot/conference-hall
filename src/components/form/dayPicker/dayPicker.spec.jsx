/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme/build'
import { DayPicker } from './dayPicker'

let onChange
let props

describe('components/dayPicker', () => {
  beforeEach(() => {
    onChange = jest.fn()
    props = {
      id: '1',
      value: '',
      isMobile: false,
      isTablet: true,
      onChange,
    }
  })

  it('should render', () => {
    expect(<DayPicker {...props} />).toMatchSnapshot()
  })

  it('should handle return key as button click', () => {
    const wrapper = shallow(<DayPicker {...props} defaultValue="" />)
    wrapper.find('DatePicker').simulate('change', new Date(0))
    expect(onChange).toHaveBeenCalled()
    expect(wrapper.state().date).toEqual(new Date(0))
  })
})
