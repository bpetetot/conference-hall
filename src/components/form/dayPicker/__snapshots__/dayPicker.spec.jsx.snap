// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/dayPicker should render 1`] = `
<DayPicker
  id="1"
  isMobile={false}
  isTablet={true}
  onChange={[MockFunction]}
  value=""
/>
`;
